### PR TITLE
[CPU] [RISC-V] add compatible headers for new version of rvv intrinsics

### PR DESCRIPTION
**Pull request by BHbean**
# Description

I have been trying to cross-compile `openvino` for RISC-V target with rv64gcv arch recently, and I found an issue related to compatiblity of rvv intrinsics. See this [issue](https://github.com/openvinotoolkit/openvino/issues/23784) for detailed information.

Fixes the following github issue
https://github.com/openvinotoolkit/openvino/issues/23784

# Checklist

## General

- [x] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [x] Have you formatted the code using clang-format?

## Performance improvements

- [ ] Have you submitted performance data that demonstrates performance improvements?

### New features

- [ ] Have you published an RFC for the new feature?
- [ ] Was the RFC approved?
- [ ] Have you added relevant tests?

### Bug fixes

- [x] Have you included information on how to reproduce the issue (either in a github issue or in this PR)?
- [ ] Have you added relevant regression tests?

## [RFC](https://github.com/oneapi-src/oneDNN/tree/rfcs) PR

- [ ] Does RFC document follow the [template](https://github.com/oneapi-src/oneDNN/blob/rfcs/rfcs/template.md#onednn-design-document-rfc)?
- [ ] Have you added a link to the rendered document?

### CODE REVIEW

File: src/cpu/rv64/rvv_nchw_pooling.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 #include "rvv_nchw_pooling.hpp"
 #include <algorithm>
-#include <riscv_vector.h>
+#include "intrin_rvv.h"
```
Comment (dzarukin) [inline src/cpu/rv64/rvv_nchw_pooling.cpp:20]:
```suggestion
#include "cpu/rv64/intrin_rvv.hpp"
```

File: src/cpu/rv64/intrin_rvv.hpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// Referenced from
```
Comment (dzarukin) [inline src/cpu/rv64/intrin_rvv.hpp:1]: Copyright banner?

File: src/cpu/rv64/intrin_rvv.hpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#if defined(__riscv_v_intrinsic) && __riscv_v_intrinsic > 10999
+#include "intrin_rvv_010_compat_non-policy.h"
+#include "intrin_rvv_010_compat_overloaded-non-policy.h"
+#endif
+
+#if defined(__riscv_v_intrinsic) && __riscv_v_intrinsic > 11999
+#include "intrin_rvv_011_compat.h"
+#endif
+
+#endif
```
Comment (dzarukin) [inline src/cpu/rv64/intrin_rvv.hpp:16]: End-of-File symbol is missing.

File: src/cpu/rv64/intrin_rvv.hpp (hunk 2 above)
Comment (BHbean) [reply src/cpu/rv64/intrin_rvv.hpp:1]: Sorry for the late reply! Could you plz tell me how to resolve the copyright issues?

**Comment by vpirogov** [comment]
The fact that this patch includes 30K LOCs of copy pasted code is a bit intimidating. I would suggest updating intrinsics in pooling implementation to [RVV intrinsics v0.11](https://github.com/riscv-non-isa/rvv-intrinsic-doc/releases/tag/v0.11.1) or even better [stable RVV intrinsics v0.12](https://github.com/riscv-non-isa/rvv-intrinsic-doc/releases/tag/v0.12.0).

**Comment by BHbean** [comment]
> The fact that this patch includes 30K LOCs of copy pasted code is a bit intimidating. I would suggest updating intrinsics in pooling implementation to [RVV intrinsics v0.11](https://github.com/riscv-non-isa/rvv-intrinsic-doc/releases/tag/v0.11.1) or even better [stable RVV intrinsics v0.12](https://github.com/riscv-non-isa/rvv-intrinsic-doc/releases/tag/v0.12.0). Compiler support is documented [here](https://github.com/riscv-non-isa/rvv-intrinsic-doc).

Thanks for your reply! I will try to update the intrinsics to the latest stable version! :)

**Comment by vpirogov** [comment]
Closing as stale.

**Pull request closed without merging**